rpa-home/modules/ROOT/pages/automation-usecase-legacysystemdataextraction-extractfromscreen.adoc

```diff
@@ -0,0 +1,97 @@
+
+image::usecase-legacysystem-screeninteraction-searchpattern-text.png[]
+
+Because it might be that the order has no corresponding tracking number in the system and you don't want to falsely return the initial value of the variable as tracking number, you check this case first using the *Managed* Action Step.
+
+In the *DoAction*, which works like a try element, you check if the result list is empty. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-noentries-image[the screen], in case of no results, an up arrow is shown directly below the empty space where the results should have been.
+
+You use the *Search Pattern* Action Step to search for this arrow up symbol and limit the search area to the space directly below the column name.
+
+If the bot doesn't find the pattern- which in your case means that the system found a corresponding tracking number - *Search Pattern* returns an error. To continue with the workflow you need to force an OK state then, which you do in the *OnError* section.
+
+image::usecase-legacysystem-screeninteraction-searchpattern-relatedcoordinates.png[]
+
+If the legacy application returns the tracking number, the number of found patterns matching the arrow up symbol is zero. You use this information as a case differentiator in the *Select Case* Action Step.
+
+image::usecase-legacysystem-screeninteraction-selectcase.png["The Select Case Action Step uses Found Matched Pattern Count as Pin Variable Found Matched Pattern Count. The case compares the value to zero."]
+
+If there is a result, you set the gateway variable `is_orderid_found` to `true`.
+
+image::usecase-legacysystem-screeninteraction-selectcase-orderfoundtrue.png[]
+
+Then you use artificial intelligence enhanced optical character recognition to extract the tracking number with the *AI OCR (screen-based)* Action Step using the search mode `Recognition Only`.
+
+image::usecase-legacysystem-screeninteraction-aiocr.png[]
+
+As before, you use the pattern `Tracking No.` as the anchor for the search area.
+
+image::usecase-legacysystem-screeninteraction-trackingno-searcharea.png[]
+
+You save the extracted tracking number to the Activity Parameter so you can return it at the end of the process.
+
+image::usecase-legacysystem-screeninteraction-getregocnizedtext.png["The text returned by the OCR Action Step is stored in the Pin Variable named Recognized Text."]
+
+If the tracking number is not returned by the search, you set the gateway variable `is_orderid_found` to `false` and force an OK state to continue the process.
+
+// Turn off section numbering
+:sectnums!:
+
+## Next Activity
+
+* xref:automation-usecase-legacysystemdataextraction-reacttodifferentoutcomes.adoc[]
+
+## See Also
+
+// Features of RPA Manager and RPA Builder used in this topic
+* xref:rpa-builder::toolbox-control-run-program.adoc[RPA Builder Action Step: Run Program]
```

Comment:
I assume you need to hard-code the link text here?

Reply:
We used this format before in the RPA docs.